### DISCUSSION TITLE
Messages held

### DISCUSSION THREAD

**Discussion by plancked** [postalserver/postal#3215] [Help with using Postal]
HI,

Since I have a new ip address which I'm trying to warm up, I applied a limit of 60 messages per 60 minutes.

I sent out a newsletter of 2000 emails. the first 60 were sent.

The remaining emails are held by Postal, but I just realised that I need to manually release each message? I thought Postal will take care of that and will automatically send after 60 minutes.

Is there an automatic way to release all emails?

Thanks
Alex

**Comment by willpower232** [comment]
I'm afraid once a message is held, it won't be sent unless someone does so manually.

I think there is a code snippet around here somewhere for sending all the messages in code or cli which probably saves you a few clicks :sweat_smile:

**Reply by plancked** [reply]
shit!
this should not be like that! what a mess.

it will take me days to go through the messages and send them manually!

**Reply by olegbliaher** [reply]
@plancked the following will work:

```
postal console
server = Server.find_by(name: 'your-postal-server-name')
server.message_db.messages(:where => {:held => 1, :scope => 'outgoing'}).each { |m| m.add_to_message_queue(:manual => true) }
```

**Reply by petr-larin** [reply]
This indeed works if I want to resend them. But if I no longer need the "held" messages, what do I do? The UI says: "It will be held until March 04, 2026 13:26." - but what happens next??
I also tried clicking "Cancel Hold" - the status flipped to HOLD CANCELED - what's next?

**Reply by willpower232** [reply]
Thats it, once the message is no longer held, it will just sit there patiently waiting to be deleted when the retention period is up
